File: ide/0.0.3/Owl/log/WriterInterface.zep.php
<?php

namespace Owl\Log;

interface WriterInterface
{

    /**
     * Set formatter for writer
     *
     * @param mixed $formatter 
     */
    public function setFormatter($formatter);

    /**
     * Get formatter
     *
     * @return FormatterInterface 
     */
    public function getFormatter();

    /**
     * Set options for writer
     *
     * @param mixed $options 
     */
    public function setOptions($options);

    /**
     * Get options for writer
     *
     * @return array 
     */
    public function getOptions();

    /**
     * Get option for writer
     *
     * @param string $option 
     */
    public function getOption($option);

    /**
     * Set message levels for writer
     *
     * @param mixed $levels 
     */
    public function setLevels($levels);

    /**
     * Get message levels for writer
     *
     * @return array 
     */
    public function getLevels();

    /**
     * Commit pack of messages
     *
     * @param array $records 
     * @return bool 
     */
    public function commit($records);

    /**
     * Push all messages
     *
     * @return bool 
     */
    public function push();

}

File: ext/owl/log/writerinterface.zep.c

#ifdef HAVE_CONFIG_H
#include "../../ext_config.h"
#endif

#include <php.h>
#include "../../php_ext.h"
#include "../../ext.h"

#include <Zend/zend_exceptions.h>

#include "kernel/main.h"


ZEPHIR_INIT_CLASS(Owl_Log_WriterInterface) {

	ZEPHIR_REGISTER_INTERFACE(Owl\\Log, WriterInterface, owl, log_writerinterface, owl_log_writerinterface_method_entry);

	return SUCCESS;

}

/**
 * Set formatter for writer
 */
ZEPHIR_DOC_METHOD(Owl_Log_WriterInterface, setFormatter);

/**
 * Get formatter
 */
ZEPHIR_DOC_METHOD(Owl_Log_WriterInterface, getFormatter);

/**
 * Set options for writer
 */
ZEPHIR_DOC_METHOD(Owl_Log_WriterInterface, setOptions);

/**
 * Get options for writer
 */
ZEPHIR_DOC_METHOD(Owl_Log_WriterInterface, getOptions);

/**
 * Set option for writer
 */
ZEPHIR_DOC_METHOD(Owl_Log_WriterInterface, setOption);

/**
 * Get option for writer
 */
ZEPHIR_DOC_METHOD(Owl_Log_WriterInterface, getOption);

/**
 * Set message levels for writer
 */
ZEPHIR_DOC_METHOD(Owl_Log_WriterInterface, setLevels);

/**
 * Get message levels for writer
 */
ZEPHIR_DOC_METHOD(Owl_Log_WriterInterface, getLevels);

/**
 * Commit pack of messages
 */
ZEPHIR_DOC_METHOD(Owl_Log_WriterInterface, commit);

/**
 * Push all messages
 */
ZEPHIR_DOC_METHOD(Owl_Log_WriterInterface, push);

